auraed/src/runtime/cells/executable.rs

```diff
@@ -1,26 +1,76 @@
+use crate::runtime::cells::ExecutableName;
 use cgroups_rs::CgroupPid;
 use log::info;
 use std::io;
 use std::process::{Child, Command, ExitStatus};
+use thiserror::Error;
+
+// TODO: I don't know how I (future-highway) feel about this now that I've spun out
+//   `ExecutableError` from `CellsError`. It allows for consuming `Command` on `start`
+//   (slight safety gain) but probably at the expense of mapping our errors to `Status`.
+//   If there are no security concerns with exposing the underlying errors, then the dev overhead
+//   of mapping to `Status` is no much, but I'm thinking under the assumption that we won't expose.
+#[derive(Error, Debug)]
+pub(crate) enum ExecutableError {
+    #[error("failed to start executable '{executable_name}' ({command:?}) due to: {source}")]
+    FailedToStart {
+        executable_name: ExecutableName,
+        command: Command,
+        source: io::Error,
+    },
+    #[error("failed to stop executable '{executable_name}' ({executable_pid:?}) due to: {source}")]
+    FailedToStopExecutable {
+        executable_name: ExecutableName,
+        executable_pid: CgroupPid,
+        source: io::Error,
+    },
+}
 
 #[derive(Debug)]
-pub(crate) struct Executable(Child);
+pub(crate) struct Executable {
+    name: ExecutableName,
+    child: Child,
+}
 
 impl Executable {
-    pub fn spawn(mut command: Command) -> io::Result<Self> {
-        let child = command.spawn()?;
-        Ok(Self(child))
+    pub fn start(
+        name: ExecutableName,
+        mut command: Command,
+    ) -> Result<Self, ExecutableError> {
+        match command.spawn() {
+            Ok(child) => Ok(Self { name, child }),
+            Err(e) => Err(ExecutableError::FailedToStart {
+                executable_name: name,
+                command,
+                source: e,
+            }),
+        }
     }
 
-    pub fn kill(mut self) -> io::Result<ExitStatus> {
-        let id = self.0.id();
-        self.0.kill()?;
-        let exit_status = self.0.wait()?;
+    pub fn kill(&mut self) -> Result<ExitStatus, ExecutableError> {
```

Comment:
should this be "stop" now?

Reply:
I was going with the assumption that there are a couple of signals we may want to send the process, and that the api would be expanded to accommodate that. 

I was initially going to make this fn private and add `pub fn stop(signal: SignalsThatStopTheExeEnum)`, which would then call the private implementations for each signal, but I didn't feel this PR was the one to tackle that in (aka I couldn't do it in the 2 minutes I gave it before getting  back to the error stuff for the PR, lol).